src/bindings/python/tests/mock/pyngraph_fe_mock_api/CMakeLists.txt

```diff
@@ -11,7 +10,26 @@ source_group("src" FILES ${PYBIND_FE_SRC})
 
 pybind11_add_module(${PYBIND_FE_NAME} MODULE ${PYBIND_FE_SRC})
 
-target_link_libraries(${PYBIND_FE_NAME} PRIVATE ${TARGET_FE_NAME} frontend_common::static)
+set(DEPENDENCIES ov_mock_py_frontend openvino::runtime)
+set(DEFINITIONS)
+
+if (ENABLE_OV_ONNX_FRONTEND)
+    list(APPEND DEPENDENCIES openvino::frontend::onnx)
+    list(APPEND DEFINITIONS ENABLE_OV_ONNX_FRONTEND)
+endif()
+
+if (ENABLE_OV_TF_FRONTEND)
+    list(APPEND DEPENDENCIES openvino::frontend::tensorflow)
+    list(APPEND DEFINITIONS ENABLE_OV_TF_FRONTEND)
+endif()
+
+if (ENABLE_OV_PADDLE_FRONTEND)
+    list(APPEND DEPENDENCIES openvino::frontend::paddle)
+    list(APPEND DEFINITIONS ENABLE_OV_PADDLE_FRONTEND)
+endif()
+
+target_compile_definitions(${PYBIND_FE_NAME} PRIVATE ${DEFINITIONS})
+target_link_libraries(${PYBIND_FE_NAME} PRIVATE ${DEPENDENCIES})
```

Comment:
Why do we need link with FEs? I thought that we need to load FE extensions in runtime.

Reply:
why do we need to load FE extensions in runtime if we can manage dependencies during cmake stage?
especially in the tests

Reply:
discussed offline.
the current situation in python, if the onnx frontend is not available 

![image](https://user-images.githubusercontent.com/21181928/149918240-2eea5e8b-5aa6-451c-ad0e-60f41a163a51.png)
@akuporos what do you think is it valid?

Reply:
I guess it's acceptable for now, I will try to improve usability in the next PR